rotkehlchen/chain/ethereum/modules/superchain_bridge/decoder.py

```diff
@@ -34,10 +41,16 @@
 WITHDRAWAL_PROVEN: Final = b'g\xa6 \x8c\xfc\xc0\x80\x1dP\xf6\xcb\xe7ds?O\xdd\xf6j\xc0\xb0DB\x06\x1a\x8a\x8c\x0c\xb6\xb6?b'  # noqa: E501
 
 
-class OptimismBridgeDecoder(DecoderInterface):
+class SuperchainBridgeDecoder(DecoderInterface):
```

Comment:
Would it not make more sense to

Make this an abstract class that is abstracted? And then use it where you need in two different implementation? As this seems to just combine two different chains from the side of ethereum, which is not what we normally do and introduced multiple if conditionals.

Just like you did with SuperchainCommon. But this time, this one should be superchainL1SideCommonDecoder or something like this as a name